inspirehep/records/api/base.py

```diff
@@ -684,3 +720,173 @@ def _add_file(self, url, original_url=None, filename=None, **kwargs):
+                f"type, but $schema says that this is {self._schema_type} type object!"
+            )
+        if delete:
+            result = InspireRecordIndexer().delete(self)
+        else:
+            result = InspireRecordIndexer().index(self)
+        logger.info(f"Indexing finished: {result}")
+        return result
+
+    def index(self, delete=None):
+        """Reindexing task sent to `index_record`
+        which calls record._index in there.
+
+        Args:
+            delete: set to True if record is deleted,
+                If not set, tries to determine this automagically
+        Returns:
+            celery.result.AsyncResult: Task itself
+        """
+        logger.error(f"Indexing record {self.id}")
+        indexing_args = self._record_index(self, deleted=delete)
+        indexing_args["record_version"] = self.model.version_id
+        task = index_record.delay(**indexing_args)
+        logger.info(f"Record {self.id} send for indexing")
+        return task
+
+    @classmethod
+    def _record_index(cls, record, _id=None, deleted=None):
+        """Helper function for indexer:
+            Prepare dictionary for indexer
+        Returns:
+            dict: proper dict required by the indexer
+        """
+        if isinstance(record, InspireRecord):
+            uuid = record.id
+        elif _id:
+            uuid = _id
+        else:
+            raise MissingModelError
+        arguments_for_indexer = {
+            "uuid": uuid,
+            "deleted": deleted or record.get("deleted", False),
+        }
+        return arguments_for_indexer
+
+    @property
+    def _previous_version(self):
+        """Allows to easily acces previous version of the record"""
+        data = (
+            self.model.versions.filter_by(version_id=self.model.version_id)
+            .one()
+            .previous.json
+        )
+        return type(self)(data=data)
+
+    @property
+    def _schema_type(self):
+        return PidStoreBase.get_pid_type_from_schema(self["$schema"])
+
+    def get_modified_references(self):
+        """Return the ids of the references diff between the latest and the
+        previous version.
+
+        The diff includes references added or deleted. Changes in a
+        reference's content won't be detected.
+
+        Also, it detects if record was deleted/un-deleted compared to the
+        previous version and, in such cases, returns the full list of
+        references.
+
+        References not linked to any record will be ignored.
+
+        Note: record should be committed to DB in order to correctly get the
+        previous version.
+
+        Returns:
+            Set[Tuple[str, int]]: pids of references changed from the previous
+            version.
+        """
+        try:
+            prev_version = self._previous_version
+        except AttributeError:
+            prev_version = {}
+
+        changed_deleted_status = self.get("deleted", False) ^ prev_version.get(
```

Comment:
I would use `and` here instead of `^`

Reply:
but this will give different result. We want to check if state of deleted field has changed between versions.
`
>True ^ True  -> False
>True ^ False  -> True
>False ^ True -> True
>False ^ False -> False
`

`
>True and True -> True
>True and False -> False
>False and True -> False
>False and False -> False
`

I could use `!=` instead of `^` maybe, but does it matter?